scheduler/src/cook/kubernetes/compute_cluster.clj

```diff
@@ -163,192 +42,120 @@
-                 :reason (-> state .getTerminated .getReason)}))
-            :default
-            {:state :pod/unknown
-             :reason "Unknown"}))
-
-        {:state :pod/waiting
-         :reason "Pending"}))))
-
-
-(defn pod-state->task-status
-  [pod {:keys [state reason]}]
-  (let [task-state (case state
-                     :pod/waiting :task-staging
-                     :pod/running :task-running
-                     :pod/succeeded :task-finished
-                     :pod/failed :task-failed)
-        task-id (-> pod .getMetadata .getName)]
-
-    {:task-id {:value task-id}
-     :reason reason ; TODO(pschorf): Map as best as possible to mesos reasons
-     :state task-state}))
-
 (defn make-pod-watch-callback
-  [handle-status-update handle-exit-code]
+  [kcc]
   (fn pod-watch-callback
     [prev-pod pod]
     (try
-      (let [prev-state (pod->pod-state prev-pod)
-            state (pod->pod-state pod)
-            task-id (-> (or pod prev-pod)
-                        .getMetadata .getName)
-            namespace (-> (or pod prev-pod)
-                          .getMetadata
-                          .getNamespace)]
-        (if (= namespace "cook")
-          (when (not (= prev-state state))
-            (log/debug "Updating state for" task-id "from" prev-state "to" state)
-            (case (:state state)
-              :pod/running
-              (handle-status-update (pod-state->task-status pod state))
-              :pod/waiting
-              (handle-status-update (pod-state->task-status pod state))
-              :pod/succeeded
-              (do
-                (handle-status-update (pod-state->task-status pod state))
-                (handle-exit-code task-id (:exit state)))
-              :pod/failed
-              (do
-                (handle-status-update (pod-state->task-status pod state))
-                (handle-exit-code task-id (:exit state)))
-              :pod/unknown
-              (log/error "Unable to determine pod status in callback")))
-          (log/debug "Skipping state update for" task-id)))
+      (if (nil? pod)
+        (controller/pod-deleted kcc prev-pod)
+        (controller/pod-update kcc pod))
       (catch Exception e
         (log/error e "Error processing status update")))))
 
-(defrecord KubernetesComputeCluster [^ApiClient api-client name entity-id match-trigger-chan exit-code-syncer-state current-pods-atom current-nodes-atom]
+(defn task-ents->map-by-task-id
+  [task-ents]
+  (->> task-ents
+       (map (fn [task-ent] [(str (:instance/task-id task-ent)) task-ent]))
+       (into {})))
+
+(defn task-ent->expected-state
+  [task-ent]
+  
+  (case (:instance/status task-ent)
+    :instance.status/unknown {:expected-state :expected/starting}
+    :instance.status/running {:expected-state :expected/running}
+    :instance.status/failed {:expected-state :expected/completed}
+    :instance.status/success {:expected-state :expected/completed}))
+
+(defn determine-expected-state
```

Comment:
Rename this determine-expected-state-on-startup

Reply:
👍